jetty-util/src/main/java/org/eclipse/jetty/util/statistic/RateCounter.java

```diff
@@ -0,0 +1,61 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.util.statistic;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.LongAdder;
+
+/**
+ * Gives the same basic functionality of {@link LongAdder} but allows you to check
+ * the rate of increase of the sum since the last call to {@link #getRate()};
+ */
+public class RateCounter
+{
+    private final LongAdder _total = new LongAdder();
```

Comment:
Is the total really necessary in this class?
It is entirely independent so no benefit being in here.  If you just want a rate you don't need it, if you also want the total then just have a LongAdder in addition to a RateCounter.

Reply:
I removed the total from the RateCounter. 

You now have to explicitly call  `RateCounter.reset()` to reset it, it doesn't do this automatically on the `RateCounter.getRate()` call anymore, the reset is called from the `ConnectionStatistics.getRate()` methods.